riscv/execute.cc

```diff
@@ -304,13 +304,15 @@ void processor_t::step(size_t n)
     }
     catch(trap_t& t)
     {
+      // save exception PC before it gets changed by take_trap
+      const reg_t exc_pc = pc;
       take_trap(t, pc);
       n = instret;
 
       // Trigger action takes priority over single step
       auto match = TM.detect_trap_match(t);
       if (match.has_value())
-        take_trigger_action(match->action, 0, state.pc);
+        take_trigger_action(match->action, 0, exc_pc);
```

Comment:
This part of the modification seems to address another issue. Perhaps, we can have another commit for this modification.